app/src/main/java/com/example/fragrecview/ui/userdetails/UserDetailsFragment.kt

```diff
@@ -34,21 +34,32 @@ class UserDetailsFragment : Fragment() {
         recyclerView = view.findViewById(R.id.recyclerViewUsers)
         noUsersTextView = view.findViewById(R.id.noUsers)
         recyclerView.layoutManager = LinearLayoutManager(requireContext())
-        rvAdapter = UserAdapter()
+        rvAdapter = UserAdapter { user ->
+            userDetailsViewModel.deleteUser(user.userID)
+            val updatedUsers = userDetailsViewModel.getUsers()
+            rvAdapter.updateData(updatedUsers)
+            updateUI(updatedUsers)
+        }
+
         recyclerView.adapter = rvAdapter
 
-        userDetailsViewModel = ViewModelProvider(this).get(UserDetailsViewModel::class.java)
+        val userDao = UserDatabase.getDatabase(requireContext()).userDao()
+        userDetailsViewModel = UserDetailsViewModel(userDao)
```

Comment:
use viewmodel provider to initialize the viewmodel. You can not initialize viewmodel like this.